Java/policy-service/src/main/java/com/insuranceapplication/policyservice/methods/Utils.java

```diff
@@ -1,23 +1,146 @@
-            list.add(mapToObject(map,clazz));
+        for (LinkedHashMap map : mapList) {
+            list.add(mapToObject(map, clazz));
         }
         return list;
     }
 
+    public static Double percentToDouble(String percentage) {
+        return Double.parseDouble(percentage.replace(",", ".").replace("%", "")) / 100;
+
+    }
+
+    public Integer yearsFromNow(Date date) {
+        if (date == null) {
+            date = new Date();
+        }
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+        return Math.abs(Period.between(LocalDate.parse(sdf.format(date)), LocalDate.now()).getYears());
+    }
+
+    public Double getDoubleFromPrecentage(String precentage, Double value) {
+        Double precentageOfBonus = Double.valueOf(precentage.replace(",", ".").replace("%", ""));
+        Double sumToAdd = value.floatValue() * (precentageOfBonus / 100.0D);
+        DecimalFormat df = new DecimalFormat("###.##", new DecimalFormatSymbols(Locale.US));
+        return Double.valueOf(df.format(sumToAdd));
+
+    }
+
+    public Customer getCustomerFromInsuredObject(InsuredObject driver) {
+        Customer custDriver = new Customer();
+        custDriver.setId(driver.getN01());
+        ResponseEntity response = template.postForEntity(eurekaClient.getApplication(Variables.dbName)
+                .getInstances().get(0).getHomePageUrl() + "/searchcustomers", custDriver, List.class);
+        ArrayList listFromResponse = (ArrayList) Utils.mapToList((List<LinkedHashMap>) response.getBody(), Customer.class);
+        return (Customer) listFromResponse.get(0);
+    }
+
+    public Vehicle getVehicleFromInsuredObject(InsuredObject insuredVehicle) {
+        Vehicle vehicle = new Vehicle();
+        vehicle.setId(insuredVehicle.getN01());
+        ResponseEntity response = template.postForEntity(eurekaClient.getApplication(Variables.dbName)
+                .getInstances().get(0).getHomePageUrl() + "/getvehicle", vehicle, Vehicle.class);
+        return (Vehicle) response.getBody();
+    }
+
+    public InsuredObject findInsuredObject(Policy policy, String objectType) {
+        ResponseEntity response = template.postForEntity(eurekaClient.getApplication(Variables.dbName)
+                .getInstances().get(0).getHomePageUrl() + "/getpolicyline", policy, PolicyLine.class);
+        PolicyLine policyLine = (PolicyLine) response.getBody();
+        List<InsuredObject> insObjects = Utils.mapToList((List<LinkedHashMap>) policyService.getInsuredObjects(policyLine), InsuredObject.class);
+        return insObjects.stream().filter(x -> x.getType().equals(objectType)).collect(Collectors.toList()).get(0);
+    }
+
+    /**
+     * helper method to get correct configuration variables.
+     *
+     * @param combinationName
+     * @param comboId
+     * @return returns configuration row of params.
+     */
+    public PremiumCalcConfigValue getCalcConfigValue(String combinationName, String comboId, String version) {
```

Comment:
Takie metody nie powinny znajdować się w klasie przechowującej metody ogólne.  To powinno zawierać się w miejscu w którym to będzie wykorzystane. Tu ewidentnie widać że będzie to wykorzystywane tylko przez klasę PremiumCalculation.